cScripts/CavFnc/functions/modules/fn_moduleCallEndex.sqf

```diff
@@ -11,15 +13,73 @@
  *
  * Public: No
  */
+ 
+//Module Options
+private _dialogResults = [
+	"Endex Options",
+	[
+		["Set Player Weapons To Safe", ["Yes", "No"], 0],
+		["Pacify AI", ["Yes", "No"], 0],
+		["Hint Players That Fire", ["Yes", "No"], 0]
+	]
+] call Ares_fnc_showChooseDialog;
 
-"ENDEX ENDEX ENDEX" remoteExecCall ["systemChat", 0];
-"DEBRIEF IN GAME" remoteExecCall ["systemChat", 0];
-"LEADERS DEBRIEF IN CHANNEL BELOW" remoteExecCall ["systemChat", 0];
+if (count _dialogResults == 0) exitWith {};
 
+//handles results and turns them into booleans
+_dialogResults params ["_weaponsSafe", "_aiPacified", "_VCSTIME"];
+
+private _weaponsSafe = _weaponsSafe isEqualTo 0;
+private _aiPacified = _aiPacified isEqualTo 0;
+private _VCSTIME = _VCSTIME isEqualTo 0;
+
+
+//systemChat Endex message
+"ENDEX ENDEX ENDEX" remoteExecCall ["systemChat", 0]; //kept 0 for server logs
+"DEBRIEF IN GAME" remoteExecCall ["systemChat", 2];
+"LEADERS DEBRIEF IN CHANNEL BELOW" remoteExecCall ["systemChat", 2];
+
+
+//hint Endex message
 private _title = "<t color='#ffc61a' size='1.2' shadow='1' shadowColor='#000000' align='center'>ENDEX ENDEX ENDEX!</t><br />";
 private _image = "<img size='5' image='cScripts\Data\Images\7CAV_LOGO_01.paa' align='center'/><br /><br />";
 private _text0 = "<t font='PuristaMedium' size='1.1'>Mission complete</t><br /><br />";
 private _text1 = "Hold your fire and report to your Squad Leaders for debriefing in game.<br />";
-private _text3 = "<t font='PuristaMedium' size='1.1'>Please safe your weapons!</t>";
+private _text3 = "<t font='PuristaMedium' size='1.1'>Please safe your weapons if you haven't already!</t>";
+
+parseText(_title + _text0 + _image + _text1 + _text3) remoteExecCall ["hint", 2];
+
+
+//Change AI to careless (doesn't affected AI created after Endex) AND Set Safety to all Players
+{
+	if (isPlayer _x) then {
+		if (_weaponsSafe) then {
+			private _weapon = currentWeapon _x; 
+			private _safedWeapons = _x getVariable ['ace_safemode_safedWeapons', []]; 
+			if !(_weapon in _safedWeapons) then {  
+				[_x, currentWeapon _x, currentMuzzle _x] call ace_safemode_fnc_lockSafety; 
+			};
+		};
+	} else {
+		if (_aiPacified) then {
+			(group _x) setBehaviourStrong "CARELESS";
+		};
+	};
+} forEach allUnits;
```

Comment:
This will lower preference some what cause this is run on all units on the mission. Even if you dont use `_aiPacified` it will still run on a AI unit consider splitting this in to two separat "functions".